src/journal-remote/journal-upload.c

```diff
@@ -203,6 +311,27 @@ int start_upload(Uploader *u,
                         return log_oom();
                 h = l;
 
+                if (arg_compression != COMPRESSION_NONE) {
+                        char *header;
```

Comment:
`header` is currently leaked. should have `_cleanup_free_` too, and NULL initialization